reinforcement_learning/include/trace_logger.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <string>
+
+#define TRACE_LOG( logger, msg ) do{  \
+    if(logger != nullptr) {           \
+      logger->log(msg);               \
+    }                                 \
+  } while(0)                          \
+
+namespace reinforcement_learning {
+  class i_trace {
+  public:
+    virtual void log(const std::string& msg) = 0;
```

Comment:
>virtual void log(const std::string& msg) = 0; [](start = 4, length = 45)

Are we not going to include trace levels?

Reply:
done.